tests/fixtures/webext-as-library/helpers.js

```diff
@@ -0,0 +1,15 @@
+const assert = require('assert');
+
+function testWebExtModuleImports(webExt) {
+  assert.deepEqual(Object.keys(webExt).sort(), ['cmd', 'main', 'util'].sort());
+  assert.deepEqual(Object.keys(webExt.util).sort(), ['logger', 'adb'].sort());
+  assert.deepEqual(
+    Object.keys(webExt.util.adb).sort(),
+    ['listADBDevices', 'listADBFirefoxAPKs'].sort(),
+  );
+  assert.equal(typeof webExt.cmd.run, 'function');
+}
+
+module.exports = {
+  testWebExtModuleImports,
+};
```

Comment:
> @rpl I was looking into implementing an interactive node repl to execute above commands and the check for then assert for the statements above. I did look into spawning a child process...

We can use that trick directly inside this helper module, no need to spawn a new process (also, [test.lib.imports.js is already spawning this fixture in a child process here](https://github.com/mozilla/web-ext/blob/3ef025b80d83e49b1d20375cd51511e537e4c340/tests/functional/test.lib.imports.js#L39-L41), and so this test is already running into its own nodejs instance and that is already making us sure that the `require.cache` property would contain the `adbkit` npm dependency only if we have loaded the `adb.js` module).

We can just add the assertion here, e.g. something like the following will do:

```js
function testWebExtModuleImports(webExt) {
  ...

  assertImportedADB({expectLoaded: false});
  assert.deepEqual(
    Object.keys(webExt.util.adb).sort(),
    ['listADBDevices', 'listADBFirefoxAPKs'].sort(),
  );
  assertImportedADB({expectLoaded: true});
}

function assertImportedADB({expectLoaded}) {
  const hasAdbDeps = Object.keys(require.cache).filter(
    (filePath) => filePath.includes('@devicefarmer/adbkit')
  ).length > 0;

  const msg = expectLoaded
    ? 'adb module should have been loaded'
    : 'adb module should not be loaded yet';

  assert.equal(hasAdbDeps, expectLoaded, msg);
}
```

This test is expected to pass successfully with the current implementation in `src/main.js`, and it should fail if we change `src/main.js` back to when it wasn't loading the `adb.js` module lazily.

(You may already figured this out, but in case you didn't try yep, you have to set the CI environment var to true to make the test.lib.imports.js test to run:`CI=true npm run test-function`, otherwise it is skipped and only run in the CI jobs, we do this because it does `npm link`/`npm unlink` the web-ext source repository to run that test scenario).

Reply:
@rpl I am so sorry for not noticing this. I feel so dumb sometimes. Thank you for guiding me the correct way.

Reply:
> (You may already figured this out, but in case you didn't try yep, you have to set the CI environment var to true to make the test.lib.imports.js test to run:CI=true npm run test-function, otherwise it is skipped and only run in the CI jobs, we do this because it does npm link/npm unlink the web-ext source repository to run that test scenario).

On my machine(mac), even if I set CI to true, `npm unlink` and `npm link` does not run as they require elevated permissions. Even when I first installed web-ext on my machine,  it required me to run `sudo npm link`.